spec/unit/puppet_pal_2pec.rb

```diff
@@ -313,6 +322,15 @@
         expect(result).to eq("a::aplan value")
       end
 
+      it 'can set variables in any scope' do
+        vars = {'a'=> 10, 'x::y' => 20}
+        result = Puppet::Pal.in_environment('pal_env', env_dir: testing_env_dir, facts: node_facts, variables: vars) do |ctx|
+          manifest = file_containing('testing.pp', "1+2+3+4+$a+$x::y")
+          ctx.evaluate_script_manifest(manifest)
+        end
+        expect(result).to eq(40)
+      end
+
```

Comment:
Needs tests for illegal variable names and illegal values.

Reply:
tests added